asapdiscovery-data/asapdiscovery/data/state_expanders/state_expander.py

```diff
@@ -0,0 +1,51 @@
+import abc
+from enum import Enum
+
+from asapdiscovery.data.schema_v2.ligand import Ligand
+from pydantic import BaseModel, Field
+
+
+class StateExpanderType(str, Enum):
+    STEREO = "STEREO"
+    TAUTOMER = "TAUTOMER"
+    PROTOMER = "PROTOMER"
+
+
+class StateExpanderBase(BaseModel):
+    input_ligands: list[Ligand] = Field(..., description="The ligand to be expanded")
+    expander_type: StateExpanderType = Field(
+        ..., description="The type of expander to use"
+    )
+
+    @abc.abstractmethod
+    def _expand(self) -> list["StateExpansion"]:
+        ...
+
+    def expand(self) -> list["StateExpansion"]:
+        return self._expand()
+
+
+class StateExpansion(BaseModel):
```

Comment:
This is awesome! I would consider having this class not be mutable so users can't accidentally break the sync on the class.